app/models/event_blacklisting.rb

```diff
@@ -0,0 +1,76 @@
+class EventBlacklisting < ActiveRecord::Base
+  default_value_for :enabled, true
+  after_save        :queue_sync_filtered_events
+  after_destroy     :queue_sync_filtered_events
+  after_destroy { |r| $audit_log.success("Event filter [#{r.event_name}] for provider #{r.provider_model} with ID [#{r.ems_id}] has been deleted by user [#{self.class.current_userid}]") }
+
+  YAML_FILE ||= Rails.root.join("db", "fixtures", "#{self.to_s.pluralize.underscore}.yml")
+
+  def self.add_per_provider_model(event_name, provider_model)
+    provider_model = provider_model.class.name if provider_model.kind_of?(ExtManagementSystem)
+    add(event_name, provider_model, nil)
+  end
+
+  def self.add_per_provider_instance(event_name, provider_obj)
+    return unless provider_obj
+    add(event_name, provider_obj.class.name, provider_obj.id)
+  end
+
+  def self.add(event_name, provider_model, ems_id)
+    _log.info("Creating event filter [#{event_name}] for provider [#{provider_model}] with ID [#{ems_id}] by user [#{current_userid}]")
+    self.create(:event_name     => event_name,
+                :provider_model => provider_model,
+                :ems_id         => ems_id
+               )
+  end
+
+  def enabled=(value)
+    return if enabled == value
+
+    super
+    _log.info("Event filter [#{event_name}] for provider #{provider_model} with ID [#{ems_id}] had enabled changed to #{value} by user [#{self.class.current_userid}]")
+  end
+
+  def self.per_model_filters(provider_model)
+    provider_model = provider_model.class.name if provider_model.kind_of?(ExtManagementSystem)
+    where(:enabled => true, :provider_model => provider_model, :ems_id => nil)
+  end
+
+  def self.per_instance_filters(provider_obj)
+    return [] if provider_obj.nil?
+    provider_obj.event_blacklistings.where(:enabled => true)
+  end
+
+  def self.seed
+    MiqRegion.my_region.lock do
+      YAML.load_file(YAML_FILE).each do |provider_model, events|
+        existings = where(:event_name => events, :provider_model => provider_model, :ems_id => nil).pluck(:event_name)
+        (events - existings).each do |e|
+          add_per_provider_model(e, provider_model)
+        end
+      end
+    end
```

Comment:
This feels wrong.  We probably need a way to specify out-of-the box defaults, and then custom modifications.  Not sure the best way to implement that though.